express-notes-api/index.js

```diff
@@ -0,0 +1,81 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+// const { parse } = require('path');
+const noteData = require('./data.json');
+
+app.use(express.json());
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+app.get('/api/notes', (req, res) => {
+  var notesArray = [];
+  for (var property in noteData.notes) {
+    notesArray.push(noteData.notes[property]);
+  }
+  res.json(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if (req.params.id < 0 || isNaN(req.params.id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!noteData.notes[req.params.id]) {
+    res.status(404).json({ error: 'cannot find note with id ' + req.params.id });
+  } else {
+    res.json(noteData.notes[req.params.id]);
+  }
+});
+
+app.post('/api/notes', (req, res) => {
+  if (!req.body) {
+    res.status(400).json({ error: 'content is a required field' });
```

Comment:
If the request has a body, but is still missing a `'content'` field (such as `{ ad: 'Shop at Green Burrito' }`), the response should still be a `400 Bad Request`.

Reply:
I can't seem to find this one so maybe this is older version. I think its fixed